server-mux.go

```diff
@@ -115,47 +158,75 @@ func (c *MuxConn) Read(b []byte) (int, error) {
 	return c.Conn.Read(b)
 }
 
+// Close the connection and decrements the WaitGroup
+func (c *MuxConn) Close() error {
```

Comment:
Are you sure that Close() is always called here? since it seems this is the only way to decrements Waitgroup